uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/work-pattern-api.js

```diff
@@ -13,7 +13,43 @@ define([
     return api.extend({
 
       /**
-       * This method returns the calendar for the given contact(s) and period,
+       * Assigns a work pattern to a contact
+       *
+       * @param {string} contactId
+       * @param {string} workPatternID
+       * @param {string} effectiveDate
+       * @param {string} effectiveEndDate
+       * @param {string} changeReason
+       * @param {object} params - additional parameters
+       * @return {Promise} Resolved with {Array} All Work Patterns
+       */
+      assignWorkPattern: function (contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params) {
```

Comment:
@deb1990 Are you sure this link is correct? It points to a method that accepts only one parameter. I don't see how as an argument in favor of having multiple parameters.

I think that the problem here is that we have both specific params (workPatternID, effectiveDate, etc) and one generic `params` that can hold everything, which gives the impression that something like this is possible:

```javascript
WorkPatternAPI.assignWorkPattern(null, null, null, null, null, {
  'contact_id': 1,
  'pattern_id': 2,
  'effective_date': '2017-01-01',
 // ...
})
```

To avoid confusions like that, we should either have only specific params or a single generic one.